allennlp/models/constituency_parser.py

```diff
@@ -305,11 +305,17 @@ def assemble_subtree(start: int, end: int):
             # This node is a leaf.
             if end - start == 1:
                 word = sentence[start]
+                pos_tag = pos_tags[start] if pos_tags is not None else "XX"
+                tree = Tree(pos_tag, [word])
                 if label is not None:
-                    tree = Tree(label, [word])
                     if pos_tags is not None:
```

Comment:
It feels like there's more nesting here now than you need.  If you do need it, great, that's fine, you just might want to double check that you still need both of these `ifs`.